src/components/Treeview/TreeviewItem.tsx

```diff
@@ -103,20 +153,34 @@ export class TreeviewItem extends CommonComponent<ITreeviewItemProps, any> {
     }
 
     @autobind
-    private _onItemSelect(item, checked, ev: React.FormEvent<HTMLElement>): void {
+    private _onItemSelect(item: ITreeviewItem, checked: boolean, event: any): void {
+        if (item.children.length > 0 && this.props.expandOnClick) {
```

Comment:
this seems dangerous, if this.props.expandOnClick is passed the items with children can not be selected?